drivers/SmartThings/samsung-audio/src/disco.lua

```diff
@@ -93,6 +93,18 @@ local function process_response(val)
   return info
 end
 
+local function check_samsung_model(val)
+  log.debug(string.format("Doing the Device Model check --> %s", val))
+   if string.find(val, 'WAM7500') or string.find(val, 'WAM6500') or string.find(val, 'WAM5500') or string.find(val, 'WAM3500') or string.find(val, 'WAM3501') or 
+      string.find(val, 'WAM1500') or string.find(val, 'WAM1501') or string.find(val, 'WAM1400') or string.find(val, 'WAM750') or string.find(val, 'WAM550') or
+      string.find(val, 'WAM350') or string.find(val, 'J8500') or string.find(val, 'J7500') or string.find(val, 'J6500') or string.find(val, 'J650') or string.find(val, 'H750') or
+      string.find(val, 'K650') or string.find(val, 'K850') or string.find(val, 'K950') or string.find(val, 'J6500R') or string.find(val, 'J7500R') or string.find(val, 'J8500R') then
```

Comment:
these model names should be stored in a list somewhere, preferably at the top of the file, and you should just loop through them and return true if there's a match

Reply:
unless there's a better pattern to use here @dljsjr

Reply:
I moved the model names to a list at the top of the file and then created a loop for matching with the model name 👍

Reply:
> unless there's a better pattern to use here @dljsjr

Seems reasonable to me